consul/src/test/scala/io/buoyant/consul/v1/HealthApiTest.scala

```diff
@@ -49,4 +55,53 @@ class HealthApiTest extends FunSuite with Awaits {
     await(api.serviceNodes("foo", consistency = Some(ConsistencyMode.Consistent)))
     assert(lastUri.contains("consistent=true"))
   }
+
+  test("Nodes without status are filtered as if `passing`") {
+    val service = stubService(nodesBuf)
+
+    val apiPassing = HealthApi(service, Set(HealthStatus.Passing))
+    val responsePassing = await(apiPassing.serviceNodes("hosted_web")).value
+
+    assert(responsePassing.size == 1)
+    assert(responsePassing.head.ServiceName == Some("hosted_web"))
+    assert(responsePassing.head.Node == Some("Sarahs-MBP-2"))
+
+    val apiCritical = HealthApi(service, Set(HealthStatus.Critical))
+    val responseCritical = await(apiCritical.serviceNodes("hosted_web")).value
+
+    assert(responseCritical.size == 0)
+  }
+
+  test("HealthApi supports filtering by health status `warning`") {
```

Comment:
Take it or leave it, but I feel like these three tests ("HealthApi supports filtering by health status {`warning`, `passing`, `critical` }") could be made DRYer, maybe like this:
```scala
def filterTest(status: HealthStatus, name: String) = 
  test(s"HealthApi supports filtering by health status `$name`") {
    val service = stubService(nodesWithChecksBuf)
    val api = HealthApi(service, Set(status))
    
    val response = await(api.serviceNodes("hosted_web")).value
    assert(response.size == 1)
    assert(response.head.ServiceName == Some("hosted_web"))
    assert(response.head.Node == Some(s"node-$name"))
    assert(response.head.Status == Some(status))
  }
```